.github/workflows/lint-with-darker-ruff.yml

```diff
@@ -0,0 +1,34 @@
+---
+name: Darker-Ruff Lint
+
+on:
+  workflow_call:
+
+env:
+  base_ref: ${{ github.event.pull_request.base.ref }}
+
+jobs:
+  darker-lint:
+    runs-on: ubuntu-latest
+    permissions:
+      pull-requests: write
+      contents: read
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+          ref: ${{ github.event.pull_request.head.ref }}
+
+      - uses: actions/setup-python@v5
+
+      - name: darker install
+        run: |
+          pip install git+https://github.com/akaihola/darker.git@master isort==5.11.4 ruff>=0.0.292
+
+      - name: Fetch base branch
+        run: |
+          git fetch origin ${base_ref}:${base_ref}
+
+      - name: Analyze Code
+        run: |
+          darker -r ${base_ref} --check --diff --lint "ruff check" --isort --color ./src
```

Comment:
Does this mean that only files in `./src` will be validated?

Reply:
@zdc  Updated  as per your comments
1. will do analysis for all folder (removed ./src filter)
2. removed isort